pkg/stub/handler.go

```diff
@@ -2,96 +2,436 @@ package stub
+
+		if h.operatorCfg == nil {
+			dockercfgSecret.DeepCopyInto(&h.registrySecret)
+			return nil
+		}
+
+		if h.operatorCfg.Spec.RegistryCredentials.Name == dockercfgSecret.Name {
+			err := h.manageDockerCfgSecret(event.Deleted, h.operatorCfg, dockercfgSecret)
+			if err != nil {
+				return err
+			}
+		}
+
+	case *v1alpha1.SamplesResource:
+		newStatus := corev1.ConditionTrue
+		var opcfg *v1alpha1.SamplesResource
+		if event.Deleted {
+			h.operatorCfg = nil
+			// "4a" in the "startup" workflow, just create default
+			// resource and set up that way
+			opcfg = &v1alpha1.SamplesResource{}
+		} else {
+			opcfg, _ = event.Object.(*v1alpha1.SamplesResource)
+		}
+
+		if h.operatorCfg != nil && h.operatorCfg.ResourceVersion == opcfg.ResourceVersion {
+			return nil
+		}
+
+		if len(opcfg.Spec.RegistryCredentials.Name) > 0 &&
+			len(h.registrySecret.ResourceVersion) > 0 &&
+			!opcfg.ConditionOccurred(v1alpha1.ImportCredentialsExists) {
+			h.manageDockerCfgSecret(false, opcfg, &h.registrySecret)
+		}
+
+		h.buildSkipFilters(opcfg)
+
+		if len(opcfg.Spec.Architectures) == 0 {
+			opcfg.Spec.Architectures = append(opcfg.Spec.Architectures, x86)
+		}
+
+		if len(opcfg.Spec.InstallType) == 0 {
+			opcfg.Spec.InstallType = v1alpha1.CentosSamplesDistribution
+		}
+
+		for _, arch := range opcfg.Spec.Architectures {
+			dir, err := h.GetBaseDir(arch, opcfg)
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesUpdateFailed, err, "error determining distro/type : %v")
+			}
+			files, err := h.filefinder.List(dir)
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesUpdateFailed, err, "error reading in content : %v")
+			}
+			err = h.processFiles(dir, files, opcfg)
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesUpdateFailed, err, "error processing content : %v")
+			}
+
 		}
-		logrus.Printf("template client %#v", tempclient)
-		imageclient, err := getImageClient(restconfig)
+
+		now := kapis.Now()
+		status := opcfg.Condition(v1alpha1.SamplesExists)
+		status.LastUpdateTime = now
+		if status.Status != newStatus {
+			status.Status = newStatus
+			status.LastTransitionTime = now
+		}
+		opcfg.ConditionUpdate(status)
+		err := h.sdkwrapper.Update(opcfg)
 		if err != nil {
-			logrus.Errorf("failed to get image client : %v", err)
-			return err
-		}
-		logrus.Printf("image client %#v", imageclient)
-		templist, err := tempclient.Templates("openshift").List(metav1.ListOptions{})
-		logrus.Printf("template list %#v", templist)
-		/*err := sdk.Create(newbusyBoxPod(o))
-		if err != nil && !errors.IsAlreadyExists(err) {
-			logrus.Errorf("failed to create busybox pod : %v", err)
-			return err
-		}*/
+			if !kerrors.IsConflict(err) {
+				return h.processError(opcfg, v1alpha1.SamplesUpdateFailed, err, "failed adding success condition to config: %v")
+			}
+			logrus.Println("got conflict error on success status update, just going to forgo this time vs. retry")
+		}
```

Comment:
same comment as the other status update block... I think all this can be wrapped inside the "if status!=newstatus" block?

Reply:
part of next push